docs/Plugin-Support.md

```diff
@@ -0,0 +1,583 @@
+
+```yaml
+  tags:
+    VP-Post-Title: post_title
+```
+
+makes sure that the `VP-Post-Title` tag contains the value from the `post_title` database field.
+
+You can alter tags in a filter `vp_entity_tags_{$entityName}` which takes four parameters: the original tags, entity in a state before the action, entity in a state after the action and the action. Similarly to before, use `vp_meta_entity_tags_{$entityName}` for meta entities with two extra parameters representing parent entity and its states.
+
+### Files to commit with an action
+
+So far we've only talked about action descriptions that are stored in commit _messages_ but what about the actual content?
+
+For database changes, VersionPress automatically commits the corresponding INI file. For example, for a `post/edit` action, a post's INI file is committed.
+
+> Side note: VersionPress stores database entities in the `wp-content/vpdb` folder as a set of INI files.
+
+This behavior is sufficient most of the time, however, some changes should commit more files. For example, when the post is an attachment, the uploaded file should also be committed. For this, the list of files to commit can be filtered using `vp_entity_files_{$entityName}` or `vp_meta_entity_files_{$entityName}`. The callback takes a list of possibly changed files, the entity in a state before the action, the entity in a state after the action and in case of meta entity, also the parent entity in states before and after the action.
+
+The array of files to commit can contain three different types of items:
+
+1. Single file corresponding to an entity:
+
+    ```php
+    [
+      'type' => 'storage-file',
+      'entity' => 'entity name, e.g., post',
+      'id' => 'VPID',
+      'parent-id' => 'VPID of parent (for meta entities)'
+    ]
+    ```
+    
+    VersionPress automatically calculates the right path to the file.
+
+2. All files of an entity type:
+    
+    ```php
+    [
+      'type' => 'all-storage-files',
+      'entity' => 'entity name, e.g., option'
+    ]    
+    ```
+
+3. Path on the filesystem:
+    
+    ```php
+    [
+      'type' => 'path',
+      'path' => 'some/path/with/wildcards/*'
+    ]    
+    ```
+
+The full example might look something like this:
+
+```php
+[
+  ['type' => 'storage-file', 'entity' => 'post', 'id' => VPID, 'parent-id' => null],
+  ['type' => 'storage-file', 'entity' => 'usermeta', 'id' => VPID, 'parent-id' => user-VPID],
+  ['type' => 'all-storage-files', 'entity' => 'option'],
+  ['type' => 'path', 'path' => '/var/www/wp/example.txt'],
+  ['type' => 'path', 'path' => '/var/www/wp/folder/*']
+]
+```
+
+### Non-database actions
+
+Some actions are not directly related to the database entities, e.g. plugin installation, WP update, etc. VersionPress provides function `vp_force_action` for these actions. VersionPress will use only action specified by parameters of this function and ignore all automatically catched. For example:
+
```

Comment:
Not only the priority – the actions are ignored entirely. Do you have any suggestion how to reword the sentence?